lib/RefBackend/Runtime/Runtime.cpp

```diff
@@ -204,16 +201,30 @@ void refbackrt::invoke(ModuleDescriptor *moduleDescriptor,
   // "explode" the unranked memref descriptors on the underlying function
   // into separate arguments for the rank and pointer-to-descriptor.
   for (int i = 0, e = inputs.size(); i < e; i++) {
-    packedInputs[2 * i] = ToVoidPtr(&inputUnrankedMemrefs[i].rank);
-    packedInputs[2 * i + 1] = ToVoidPtr(&inputUnrankedMemrefs[i].descriptor);
+    auto idx = 2 * i;
+    if (inputs[i].isTensor()) {
```

Comment:
we can't just trust the user inputs if they are tensors or scalars. we need to check them against the descriptors we emitted in the compiler. (this function needs to change to return `refbackrt::LogicalResult` frominclude/npcomp/RefBackend/Runtime/Support.h)

(or we can have a separate function that validates the inputs or something)